In [1]:
!pip install -qq pandas

# Objetivo 

Conocer el dataset, evaluar la calidad de los datos, analizar la distribución de variables, detectar patrones interesantes y formular hipótesis para guiar los siguientes pasos.

Insights:

* Muchas columnas tienen valores faltantes y la forma en cómo se van a imputar es llenándolas con la mediana de la colonia a la que pertenece el edificio. 
* Muchas columnas tienen outliers tambien, por lo que se les hará un tratamiento para normalizarlas o limitar sus rangos.

In [2]:
import pandas as pd

In [3]:
df = pd.read_csv("../data/raw/cuahutemoc_properties.csv")

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4513 entries, 0 to 4512
Data columns (total 18 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   property_id           4513 non-null   object 
 1   listing_type          4513 non-null   object 
 2   property_type         4513 non-null   object 
 3   url_ad                4513 non-null   object 
 4   price                 4513 non-null   float64
 5   terrain_surface       396 non-null    float64
 6   construction_surface  4502 non-null   float64
 7   num_bathrooms         4436 non-null   float64
 8   num_parking_lots      3647 non-null   float64
 9   num_bedrooms          4422 non-null   float64
 10  built_year            3983 non-null   float64
 11  conservation_status   3841 non-null   float64
 12  latitude              4513 non-null   float64
 13  longitude             4513 non-null   float64
 14  id_neighborhood       4513 non-null   int64  
 15  neighborhood         

In [9]:
df.shape

(4513, 18)

In [5]:
df.head(5)

property_id listing_type property_type  \
0  5ae456983ab717ec68e1ff2e0689c5d8     for-sale     apartment   
1  fff8db1482a59a1c9093f9ab3ac50ddf     for-sale     apartment   
2  7f0478d4f668aaf45ed3511523746fce     for-sale     apartment   
3  4a0034cf7bfdd9b610bcb4a2e5096cf0     for-sale     apartment   
4  afdc70d7a521ba91cc026062653fe841     for-sale     apartment   

                                              url_ad       price  \
0  https://rentahouse.com.mx/departamento_en_vent...   9348347.0   
1  https://www.easybroker.com/mx/listings/departa...  16200000.0   
2  https://www.vivanuncios.com.mx/a-venta-departa...   5400000.0   
3  https://www.clau.com/propiedades/3pm-departame...   6700000.0   
4  https://degohouse.com.mx/apartamento-venta-non...   3778408.0   

   terrain_surface  construction_surface  num_bathrooms  num_parking_lots  \
0              NaN                  96.0            2.0               NaN   
1              NaN                  90.0            2.0               1.0   
2              NaN                 110.0            1.0               NaN   
3              NaN                  79.0            2.0               0.0   
4              NaN                  67.0            2.0               1.0   

   num_bedrooms  built_year  conservation_status   latitude  longitude  \
0           1.0      2024.0             0.989510  19.430419 -99.163761   
1           2.0      2025.0             0.981774  19.425804 -99.167429   
2           2.0      1999.0             0.916711  19.426753 -99.161845   
3           2.0      2000.0             0.873144  19.422017 -99.164724   
4           2.0      2019.0             0.988560  19.432608 -99.133208   

   id_neighborhood neighborhood  has_garden  has_gym  
0            11817   CUAUHTEMOC         1.0      NaN  
1            11818       JUAREZ         NaN      1.0  
2            11818       JUAREZ         NaN      NaN  
3            11819   ROMA NORTE         NaN      NaN  
4            11800       CENTRO         NaN      NaN

In [11]:
df.describe().T

count          mean           std            min  \
price                 4513.0  7.142255e+06  6.278022e+06  850000.000000   
terrain_surface        396.0  2.245202e+02  1.447109e+02      32.000000   
construction_surface  4502.0  1.121617e+02  9.371629e+01      25.000000   
num_bathrooms         4436.0  1.749549e+00  7.977914e-01       1.000000   
num_parking_lots      3647.0  1.298053e+00  8.381038e-01       0.000000   
num_bedrooms          4422.0  2.085256e+00  9.508218e-01       0.000000   
built_year            3983.0  2.013244e+03  1.740734e+01    1950.000000   
conservation_status   3841.0  8.762451e-01  1.364496e-01       0.248026   
latitude              4513.0  1.942436e+01  1.467602e-02      19.400000   
longitude             4513.0 -9.915953e+01  1.246362e-02     -99.183556   
id_neighborhood       4513.0  1.181371e+04  8.303129e+00   11800.000000   
has_garden            1166.0  1.000000e+00  0.000000e+00       1.000000   
has_gym               1440.0  1.000000e+00  0.000000e+00       1.000000   

                               25%           50%           75%           max  
price                 3.500000e+06  5.500000e+06  8.350000e+06  8.500000e+07  
terrain_surface       1.320000e+02  1.910000e+02  2.735000e+02  8.220000e+02  
construction_surface  6.000000e+01  8.600000e+01  1.280000e+02  1.124000e+03  
num_bathrooms         1.000000e+00  2.000000e+00  2.000000e+00  9.000000e+00  
num_parking_lots      1.000000e+00  1.000000e+00  2.000000e+00  1.000000e+01  
num_bedrooms          2.000000e+00  2.000000e+00  2.000000e+00  9.000000e+00  
built_year            2.009000e+03  2.023000e+03  2.025000e+03  2.028000e+03  
conservation_status   8.357713e-01  9.385427e-01  9.704696e-01  9.924964e-01  
latitude              1.941242e+01  1.942140e+01  1.943443e+01  1.946461e+01  
longitude            -9.916855e+01 -9.916163e+01 -9.915367e+01 -9.912310e+01  
id_neighborhood       1.180400e+04  1.181700e+04  1.181900e+04  1.183300e+04  
has_garden            1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00  
has_gym               1.000000e+00  1.000000e+00  1.000000e+00  1.000000e+00

Observations:

*  Precio: Hay una gran asimetría a la derecha (precios muy altos que elevan el promedio). El 75% de las propiedades valen menos de $8.35 millones, pero el valor máximo ($85M) sugiere la presencia de outliers (propiedades de lujo o muy comerciales).

* Terreno: Solo hay 396 registros (de 4513), es decir, muchos datos faltantes (~91% de los datos tienen terreno desconocido).

* Construcción: La mayoría de las propiedades parecen ser departamentos o casas medianas, pero el valor máximo es muy alto (posiblemente edificios).

* Muchas propiedades tienen pocos baños y cuartos → departamentos en zonas urbanas 

* La mayoría de las propiedades son nuevas o recientes. Hay propiedades desde 1950, pero la mediana sugiere que al menos la mitad son muy nuevas

In [12]:
df.describe(include='object')

property_id listing_type property_type  \
count                               4513         4513          4513   
unique                              4513            1             2   
top     5ae456983ab717ec68e1ff2e0689c5d8     for-sale     apartment   
freq                                   1         4513          4067   

                                                   url_ad neighborhood  
count                                                4513         4513  
unique                                               4513           34  
top     https://rentahouse.com.mx/departamento_en_vent...   ROMA NORTE  
freq                                                    1          824

Observations: 

* Cada propiedad tiene un ID único, no hay duplicados

* Todas las propiedades son ventas (for-sale)

* Predominan los departamentos, con una minoría de casas u otro tipo de inmueble (446 unidades restantes ≈ 10%) → En línea con el insight anterior sobre construcciones pequeñas y urbanas.

* Cada propiedad tiene su URL única → Puedo usar esta columna para web scraping, análisis de imágenes, o inspección manual

* Hay 34 colonias, pero Roma Norte concentra el 18% de las propiedades. Otras colonias tendrán muchas menos propiedades por lo que puede causar desequilibrio en modelos o análisis de localización.



In [10]:
df.listing_type.unique()

array(['for-sale'], dtype=object)

In [7]:
df.property_type.unique()

array(['apartment', 'house'], dtype=object)

In [8]:
df.neighborhood.unique()

array(['CUAUHTEMOC', 'JUAREZ', 'ROMA NORTE', 'CENTRO', 'DOCTORES',
       'CONDESA', 'HIPODROMO', 'SAN RAFAEL', 'SANTA MARIA LA RIBERA',
       'SAN SIMON TOLNAHUAC', 'GUERRERO', 'PERALVILLO', 'MORELOS',
       'HIPODROMO DE LA CONDESA', 'VISTA ALEGRE', 'TABACALERA',
       'ROMA SUR', 'BUENOS AIRES', 'AMPL ASTURIAS',
       'UNIDAD HAB NONOALCO TLATELOLCO', 'ASTURIAS', 'ATLAMPA',
       'PAULINO NAVARRO', 'BUENAVISTA', 'ALGARIN',
       'CENTRO URBANO BENITO JUAREZ', 'VALLE GOMEZ', 'OBRERA',
       'ESPERANZA', 'TRANSITO', 'EX HIPODROMO DE PERALVILLO',
       'FELIPE PESCADOR', 'SANTA MARIA INSURGENTES', 'MAZA'], dtype=object)

### Conclusiones clave

1. Outliers en price (hasta 85M): normalizar o limitar el rango para evitar comparaciones distorsionadas.

2. Predominan departamentos (~90%): comparar propiedades del mismo tipo (property_type).

3. Colonias muy desbalanceadas: ROMA NORTE domina. Comparar solo dentro de la misma colonia o vecinas.

4. Recámaras, baños y estacionamientos están centrados en 2-2-1: estas variables son útiles para filtrar propiedades similares.

5. Año de construcción promedio reciente (~2013): usarlo para comparar propiedades de edad similar.

6. Muchos valores faltantes en terrain_surface y amenidades: priorizar construction_surface para comparabilidad.